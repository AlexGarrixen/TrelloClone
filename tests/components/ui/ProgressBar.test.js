import React from 'react';
import { mount } from 'enzyme';
import ProgressBarBar from '../../../src/components/ui/ProgressBar';

describe('<ProgressBar />', () => {
  test('Render success', () => {
    const wrapper = mount(<ProgressBarBar />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.progress-bar').exists()).toBe(true);
    expect(wrapper.find('.progress-bar__bar').exists()).toBe(true);
  });

  test('Test with prop percentage', () => {
    const wrapper = mount(<ProgressBarBar percentage={70} />);
    const elementBar = wrapper.find('.progress-bar__bar');

    expect(elementBar.prop('style').width).toBe('70%');
  });
});
