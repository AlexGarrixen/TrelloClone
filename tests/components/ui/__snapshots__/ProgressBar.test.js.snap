// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressBar /> Render success 1`] = `
<div
  className="progress-bar"
>
  <div
    className="progress-bar__bar"
    style={
      Object {
        "width": "0%",
      }
    }
  />
</div>
`;
